python_modules/dagster/dagster/_utils/schedules.py

```diff
@@ -83,6 +76,28 @@ def cron_string_iterator(
     if is_numeric[1]:
         expected_hour = int(cron_parts[1][0])
 
+    if delta_fn is not None and start_offset == 0 and croniter.match(cron_string, start_timestamp):
```

Comment:
here's my confusion - croniter.match literally calls get_prev and get_current! https://github.com/kiorky/croniter/blob/master/src/croniter/croniter.py#L813-L820

But that corresponds to your "without hacky shim" case above right?

Reply:
good callout -- dug more into the profiling, and there's two things going on.

first, in the original code, we were calling `croniter.match` anyway, so removing extra calls to get_prev/get_next will always result in a speed improvement.

second (perhaps more importantly), the time taken to get the previous tick is correlated to the distance between those ticks. Inside of `match`, they're calculating the previous tick of the schedule, relative to 1 microsecond after the input time (so the previous tick will only be one microsecond away. Inside the other codepath, we're getting the true previous tick, which is, in my example, 15 minutes away. You can see the difference in time spent in this profile:

<img width="1033" alt="Screenshot 2023-04-26 at 10 25 57 AM" src="https://user-images.githubusercontent.com/22457492/234654945-df186710-6784-4952-b286-f2349be5a1b9.png">

The left hand side is the time spent on "get_prev" in the old codepath, whereas the right hand side is the time spent on get_prev in the "match" codepath.

If you dig in even further, the fast codepath calls `proc_second` inside of `_calc`, whereas the slow codepath calls `proc_minute`.

Reply:
ahhh very interesting. thanks for digging into that